cpp/ql/src/semmle/code/cpp/models/implementations/Strdup.qll

```diff
@@ -43,3 +43,32 @@ class StrdupFunction extends AllocationFunction, ArrayFunction, DataFlowFunction
     output.isReturnValueDeref()
   }
 }
+
+/**
+ * A `strndup` style allocation function.
+ */
+class StrndupFunction extends AllocationFunction, ArrayFunction, TaintFunction {
+  StrndupFunction() {
+    exists(string name |
+      hasGlobalOrStdName(name) and
```

Comment:
I learned only yesterday that `strdup` (and `strndup`) are POSIX functions rather than C/C++ standard functions, so they can't appear in `std`. They [will be in C20](https://en.cppreference.com/w/c/experimental/dynamic/strdup) but apparently not in C++20.

Reply:
Well spotted.  I've fixed this (for `strdup`, `strndup` and `wcsdup`).